packages/integration-tests/test/actions.spec.ts

```diff
@@ -65,27 +72,113 @@ const deposit = async (amount, value) => {
     watcher = await initWatcher()
   })
 
   it('should deploy the simple storage contract', async () => {
-    SimpleStorage = await deploySimpleStorage()
+    simpleStorage = await simpleStorageFactory.deploy()
+    await simpleStorage.deployTransaction.wait()
   })
 
   it('should deposit from L1->L2', async () => {
     const value = `0x${'42'.repeat(32)}`
     await deposit(1, value)
-    const msgSender = await SimpleStorage.msgSender()
-    const l1ToL2Sender = await SimpleStorage.l1ToL2Sender()
-    const storageVal = await SimpleStorage.value()
-    const count = await SimpleStorage.totalCount()
+    const msgSender = await simpleStorage.msgSender()
+    const l1ToL2Sender = await simpleStorage.l1ToL2Sender()
+    const storageVal = await simpleStorage.value()
+    const count = await simpleStorage.totalCount()
 
     msgSender.should.be.eq(l2MessengerAddress)
     l1ToL2Sender.should.be.eq(l1Wallet.address)
     storageVal.should.be.eq(value)
     count.toNumber().should.be.eq(1)
   })
+
+  // TODO: Enable withdrawal test
+  it.skip('should withdraw from L2->L1', async () => {
+    const value = `0x${'77'.repeat(32)}`
+    await withdraw(value)
+    const msgSender = await simpleStorage.msgSender()
+    const l2ToL1Sender = await simpleStorage.l2ToL1Sender()
+    const storageVal = await simpleStorage.value()
+    const count = await simpleStorage.totalCount()
+
+    msgSender.should.be.eq(l1MessengerAddress)
+    l2ToL1Sender.should.be.eq(l2Wallet.address)
+    storageVal.should.be.eq(value)
+    count.toNumber().should.be.eq(1)
+  })
+})
+
+describe('ERC20', async () => {
+  const alice = new Wallet(etherbase, l2Provider)
+  const INITIAL_AMOUNT = 1000
+  const NAME = 'OVM Test'
+  const DECIMALS = 8
+  const SYMBOL = 'OVM'
+
+  it('should deploy the erc20 contract', async () => {
+    erc20 = await ERC20Factory.deploy(
+      INITIAL_AMOUNT, NAME, DECIMALS, SYMBOL
+    )
+  })
+
+  it('should set the total supply', async () => {
+    const totalSupply = await erc20.totalSupply()
+    expect(totalSupply.toNumber()).to.equal(INITIAL_AMOUNT)
+  })
+
+  it('should get the token name', async () => {
+    const name = await erc20.name()
+    expect(name).to.equal(NAME)
+  })
+
+  it('should get the token decimals', async () => {
+    const decimals = await erc20.decimals()
+    expect(decimals).to.equal(DECIMALS)
+  })
+
+  it('should get the token symbol', async () => {
+    const symbol = await erc20.symbol()
+    expect(symbol).to.equal(SYMBOL)
+  })
+
+  it('should assign initial balance', async () => {
+    const balance = await erc20.balanceOf(l2Wallet.address)
+    expect(balance.toNumber()).to.equal(INITIAL_AMOUNT)
+  })
+
+  it('should transfer amount to destination account', async () => {
+    const transfer = await erc20.transfer(alice.address, 100)
+    const receipt = await transfer.wait()
+    const transferLog = receipt.events.find((element) => element.event.match('Transfer'));
```

Comment:
Does this emit more than 1 event?

Reply:
It returns two `Transfer` events. One with `logIndex: 0` and one with `logIndex: 1`. Do you have any ideas?

```
[ { transactionIndex: 0,
    blockNumber: 11,
    transactionHash:
     '0x1a6b0f3e8ecf334eeb1a7282db1c33975cf4b364069ecf3832621f0682ce48a0',
    address: '0x4200000000000000000000000000000000000006',
    topics:
     [ '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
       '0x000000000000000000000000023ffdc1530468eb8c8eebc3e38380b5bc19cc5d',
       '0x0000000000000000000000004200000000000000000000000000000000000005' ],
    data:
     '0x0000000000000000000000000000000000000000000000000000000000000000',
    logIndex: 0,
    blockHash:
     '0x45a6eda785c8e056c6f1fbd123d3d03f2820f2fcd35ae4938bea421de87656e0',
    args:
     [ '0x023fFdC1530468eb8c8EEbC3e38380b5bc19Cc5d',
       '0x4200000000000000000000000000000000000005',
       [BigNumber],
       _from: '0x023fFdC1530468eb8c8EEbC3e38380b5bc19Cc5d',
       _to: '0x4200000000000000000000000000000000000005',
       _value: [BigNumber] ],
    decode: [Function],
    event: 'Transfer',
    eventSignature: 'Transfer(address,address,uint256)',
    removeListener: [Function],
    getBlock: [Function],
    getTransaction: [Function],
    getTransactionReceipt: [Function] },
  { transactionIndex: 0,
    blockNumber: 11,
    transactionHash:
     '0x1a6b0f3e8ecf334eeb1a7282db1c33975cf4b364069ecf3832621f0682ce48a0',
    address: '0x6418E5Da52A3d7543d393ADD3Fa98B0795d27736',
    topics:
     [ '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
       '0x000000000000000000000000023ffdc1530468eb8c8eebc3e38380b5bc19cc5d',
       '0x0000000000000000000000000a4984541f192294ca4cca6788519204b45d67ea' ],
    data:
     '0x0000000000000000000000000000000000000000000000000000000000000064',
    logIndex: 1,
    blockHash:
     '0x45a6eda785c8e056c6f1fbd123d3d03f2820f2fcd35ae4938bea421de87656e0',
    args:
     [ '0x023fFdC1530468eb8c8EEbC3e38380b5bc19Cc5d',
       '0x0A4984541f192294cA4cCa6788519204B45d67eA',
       [BigNumber],
       _from: '0x023fFdC1530468eb8c8EEbC3e38380b5bc19Cc5d',
       _to: '0x0A4984541f192294cA4cCa6788519204B45d67eA',
       _value: [BigNumber] ],
    decode: [Function],
    event: 'Transfer',
    eventSignature: 'Transfer(address,address,uint256)',
    removeListener: [Function],
    getBlock: [Function],
    getTransaction: [Function],
    getTransactionReceipt: [Function] } ]
```

Reply:
One of the transfer events is the transfer of ETH to pay fee (this will be a transfer of value 0) see https://github.com/ethereum-optimism/contracts-v2/blob/d5a17efac0719a3645e41599272a28a099b1ee0b/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol#L90-L95